src/libraries/Common/tests/System/Net/Http/HttpClientHandlerTest.ServerCertificates.cs

```diff
@@ -93,6 +93,18 @@ public async Task NoCallback_ValidCertificate_SuccessAndExpectedPropertyBehavior
             }
         }
 
+        [OuterLoop("Uses external servers")]
```

Comment:
You can possibly just add another entry to `DefaultConnect_EndToEnd_Ok` (or some other test)  in System.Net.Security since the fix is in SslStream. It feels like we can add HTTP test _if_ we choose to update header processing.

Reply:
`DefaultConnect_EndToEnd_Ok` doesn't reproduce the issue as it is, the test passed without this change as well. I will try to adapt this test

Reply:
that is interesting. We should figure out why. I briefly look and while it registers custom validation callback it assert that there are no validation errors. Without the fix you _should_ still get name mismatch.

Reply:
My bad, sorry, I used old dll (building test didn't update it). I added this case into `DefaultConnect_EndToEnd_Ok` and removed test from HttpClient